integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/NextMoveServiceBus.java

```diff
@@ -133,30 +133,34 @@ public void getAllMessages() throws NextMoveException {
         opts.setReceiveMode(ReceiveMode.PEEK_LOCK);
 
         ServiceBusContract service = createContract();
-        ArrayList<BrokeredMessage> messages = Lists.newArrayList();
-        for (int i=0; i<props.getNextbest().getServiceBus().getReadMaxMessages(); i++) {
-            try {
-                BrokeredMessage msg = service.receiveQueueMessage(queuePath, opts).getValue();
-                if (msg == null || isNullOrEmpty(msg.getMessageId())) {
-                    break;
+        boolean messagesInQueue = true;
+        while (messagesInQueue) {
+            ArrayList<BrokeredMessage> messages = Lists.newArrayList();
+            for (int i=0; i<props.getNextbest().getServiceBus().getReadMaxMessages(); i++) {
+                try {
+                    BrokeredMessage msg = service.receiveQueueMessage(queuePath, opts).getValue();
+                    if (msg == null || isNullOrEmpty(msg.getMessageId())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS134)